Update dependencies versions

peterdemin opened the pull request:
##### SUMMARY
Run pip-compile to update dependencies versions

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - API
 - UI
 - Installer

I noticed that most of the locked dependencies are outdated so I updated them following instructions from `requirements/README.md`.

wenottingham commented:
Looks like this broke the build? 

This at least needs to pass CI before we'd consider merging this.

peterdemin commented:
Well, that's expected, since I upgraded a few major releases.
I'll check which ones are breaking the build and pin them in `.in` file.

peterdemin commented:
Hmm... I could use some help here.
Unit tests are failing because logging can't be configured:
```
...
  File "/venv/awx/lib/python2.7/site-packages/pytest_django/plugin.py", line 246, in pytest_load_initial_conftests
    _setup_django()
  File "/venv/awx/lib/python2.7/site-packages/pytest_django/plugin.py", line 149, in _setup_django
    django.setup()
  File "/venv/awx/lib/python2.7/site-packages/django/__init__.py", line 22, in setup
    configure_logging(settings.LOGGING_CONFIG, settings.LOGGING)
  File "/venv/awx/lib/python2.7/site-packages/django/utils/log.py", line 75, in configure_logging
    logging_config_func(logging_settings)
  File "/usr/lib64/python2.7/logging/config.py", line 803, in dictConfig
    dictConfigClass(config).configure()
  File "/usr/lib64/python2.7/logging/config.py", line 585, in configure
    '%r: %s' % (name, e))
ValueError: Unable to configure handler 'console': Cannot resolve 'awx.main.utils.handlers.ColorHandler': No module named certifi
```
I checked `awx/main/utils/handlers.py`, there is no import of `certifi`. Also, this package is present in `requirements.txt`, so I don't understand, why is there a problem...
Django messes up with exception traceback during settings load, so it might be something completely different.

Commit:
Constrain requests

Commit:
Recompile dependencies with --upgrade

pip-compile --allow-unsafe --upgrade \
    --output-file requirements/requirements.txt \
    requirements/requirements.in

pip-compile --allow-unsafe --upgrade \
    --output-file requirements/requirements_ansible.txt \
    requirements/requirements_ansible.in

peterdemin commented:
I spun up docker container with awx locally and discovered that the cause is the new version of requests, that relies on certifi package, which is uninstalled to solve some 2016 cloud issues.
So I constrained requests before it introduced this dependency and recompiled `.txt` files.

cchurch commented on requirements/requirements_ansible.txt:
```diff
@@ -4,124 +4,130 @@
+debtcollector==1.19.0     # via oslo.config, oslo.utils, python-designateclient, python-keystoneclient, python-neutronclient
+decorator==4.2.1          # via openstacksdk, shade
+deprecation==1.1          # via openstacksdk
+docutils==0.14            # via botocore
```
The `docutils` line needs to be commented, since it will be installed instead from `requirements_ansible_git.txt`.

peterdemin replied:
Thanks. Will do!

peterdemin replied:
Hey @cchurch, I fixed the issue, but GitHub says, that change is still requested. Am I missing something?

cchurch reviewed (changes requested):
Needs one change to fix failing shippable test: https://app.shippable.com/github/ansible/awx/runs/1609/1/console

cchurch commented:
Additional thoughts, slightly outside the scope of this PR...


I would think that most of the packages in the various `.in` files should be specified without an explicit version to allow `pip-compile` to check for updates to those dependencies. Certain core packages should probably still specify a [compatible release](https://www.python.org/dev/peps/pep-0440/#compatible-release) version, e.g. `Django~=1.11.0` or `djangorestframework~=3.7.0` to allow for patch updates but not major/minor version changes.

peterdemin commented:
Right, that's the next step. I wanted to upgrade indirect dependencies first, so there will be fewer changes in the second PR.

peterdemin committed:
Comment out docutils from ansible requirements

Because it will be installed instead from requirements_ansible_git.txt

Signed-off-by: Peter Demin <kibitzrrr@gmail.com>

matburt commented:
Just so you know, before we merge this we're going to want to see extensive examples of various functionality passing.

Some examples:
boto[3] - ec2 modules and inventory sync still work correctly
os- keystone, neutron - openstack modules and inventory sync still work correctly
kombu, celery - message bus, job execution, job data payloads, websockets all work
python-ldap - our ldap and django interfaces still work
websocket-client - slack notifications still work correctly
**ETC**

If something major is wrong and this gets merged then this will have the effect of choking off any other development work that's going on at the moment.

Usually this work is also done in parallel with updating our ansible inventory scripts:
https://github.com/ansible/awx/tree/devel/awx/plugins/inventory

These things are not covered by our unit tests but usually go through our manual and automated functional testing.

peterdemin commented:
Hi Matthew, when you say "we're going to want to see extensive examples" do you mean, that I need to manually test those cases? Or someone else will be assigned to it?

matburt commented:
It would have to happen before this PR merges. If someone else ends up doing it then we'll likely close this PR and open another one. Dependency upgrades like this are never this straightforward.

AlanCoding commented:
I'm going to be upgrading other dependencies. What I'll probably do is that I'll finish the changes I'm making, and then I'll cherry-pick from your commits here before I complete my final run of the pip-compile command.

After I have that up, I will close this PR, but I'll leave it open until then.
